internal/controller/reinstall/reinstall.go

```diff
@@ -119,9 +119,16 @@ func (r *ReinstallHandler) ProcessRequest(
 
 	defer func() {
 		patch := client.MergeFrom(clusterInstance.DeepCopy())
-		if updateEndtime {
+		if updateEndtime && err != nil {
 			clusterInstance.Status.Reinstall.RequestEndTime = metav1.Now()
-			// Add hold annotation for manual intervention (CNF-15719)
+			// Add hold annotation for manual intervention
+			result1, updateErr := ci.SetHoldReconciliation(ctx, r.Client, log, clusterInstance, err.Error())
```

Comment:
I wonder if doing this (which I imagine actually submits an API call to alter `clusterInstance` between defining `patch` and doing the `Patch` call will cause issues?

Reply:
I haven't observed any issues between defining `patch` and executing the `Patch` call. However, in my e2e tests, I have observed that status updates to `clusterInstance` post the `ci.SetHoldReconciliation` are ignored - probably due to a change in the original `resourceVersion` of the passed `clusterInstance`. I will separate the `SetHoldReconciliation` to a separate handler with its own `client.Client` object.